Basic Calculator View

mmcdermott opened the pull request:
Sets up the Calculator show page and the necessary associated backend. No creation just yet.
![calculator-view presubmit](https://cloud.githubusercontent.com/assets/470751/16250587/5caabd98-37d3-11e6-8bf1-097d82d19aff.png)
![calc view submitted](https://cloud.githubusercontent.com/assets/470751/16250589/5fa50fe4-37d3-11e6-87da-21016e94ddb9.png)

OAGr commented on src/components/calculators/calculator-space-selector.js:
```diff
@@ -0,0 +1,52 @@
+  if (edges.inputs.length) { return OUTPUT }
+  if (edges.outputs.length) { return INPUT }
+  return NOEDGE
+}
```
Much of this code is in another selector, it would be nice to clean up.

OAGr commented on src/lib/engine/calculator.js:
```diff
@@ -0,0 +1,3 @@
+export function fullUrl({id}) {
+  return `https://www.getguesstimate.com/calculators/${id}`
```
I feel like the absolute base url should be somewhere else, this seems quite messy to me.

mmcdermott commented:
Addressed comments, and re-staged to stage.getguesstimate.com. Note that with the BASE_URL change, facebook share crashes locally (but works on stage).
